core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1397,77 +1397,172 @@ class KafkaApis(val requestChannel: RequestChannel,
     requestHelper.sendResponseMaybeThrottle(request, createResponse)
   }
 
-  private def handleOffsetFetchRequestBetweenV1AndV7(request: RequestChannel.Request): Unit = {
-    val header = request.header
+  private def handleOffsetFetchRequestBetweenV1AndV7(request: RequestChannel.Request): CompletableFuture[Unit] = {
```

Comment:
These two methods are a lot cleaner now. Is the main difference how the requests are formatted?
Ie, the v8 request is structured by group?

Reply:
I am not sure. We had a lot of complex logic before. I have tried to simplify as much as possible.